src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -236,10 +237,29 @@ private async Task<LanguageServer.Protocol.Diagnostic[]> GetDiagnosticsAsync(Doc
                 Code = diagnostic.Id,
                 Message = diagnostic.Message,
                 Severity = ProtocolConversions.DiagnosticSeverityToLspDiagnositcSeverity(diagnostic.Severity),
-                Range = ProtocolConversions.TextSpanToRange(DiagnosticData.GetExistingOrCalculatedTextSpan(diagnostic.DataLocation, text), text),
+                Range = GetDiagnosticRange(diagnostic.DataLocation, text),
                 // Only the unnecessary diagnostic tag is currently supported via LSP.
                 Tags = diagnostic.CustomTags.Contains("Unnecessary") ? new DiagnosticTag[] { DiagnosticTag.Unnecessary } : Array.Empty<DiagnosticTag>()
             }).ToArray();
         }
+
+        private LanguageServer.Protocol.Range? GetDiagnosticRange(DiagnosticDataLocation? diagnosticDataLocation, SourceText text)
+        {
+            (var startLine, var endLine) = DiagnosticData.GetLinePositions(diagnosticDataLocation, text, useMapped: true);
```

Comment:
Out of curiosity why isn't `useMapped: true` the default if it falls back to non-mapped locations in the scenario mapped ones don't exist

Reply:
It's a good question- I'm not sure what the answer is.  I dug a bit today but couldn't find anything to indicate it shouldn't be the default, but I wanted to avoid possibly changing the behavior in non-lsp scenarios so I introduced the flag.  I'll look a bit more tomorrow.